Optimise

mcmenaminadrian opened the pull request:
Key thing here is get rid of the control buffer - it seems to work but please do check it over as you understand this code better than me

mcmenaminadrian committed:
Use & and not % with rand() where we can

mcmenaminadrian committed:
Further small optimisation

mcmenaminadrian committed:
Get rid of control buffer

mcmenaminadrian commented:
I've also added two new bread types to be true to the motherland

mcmenaminadrian committed:
Update bread.c

mcmenaminadrian committed:
Update bread.c

add a (frozen) waffle

mcmenaminadrian committed:
Change waffle name to avoid scrolling issues

ConorMcPersonal commented on SplitFiles/bread.c:
```diff
@@ -55,9 +55,9 @@ BreadState* get_bread(BreadBin* bin, unsigned char typeLetter) {
     BreadState* newBread = malloc(sizeof(BreadState));
     newBread->temperature = 0;
-    newBread->moisture = type->moisture + (rand() % type->moistureVariability);
+    newBread->moisture = type->moisture + (rand() & type->moistureVariability);
```
The big drag on bread creation is the sound - saving a couple of cycles here at the expense of imposing a hidden constraint on the code (variability must be a power of 2 - 1) doesn't feel like a great trade-off.  This late in the day it's fine I suppose, but this feels like change for its own sake.

mcmenaminadrian replied:
Any sort of division eats cycles on any of these older machines and modular arithmetic is even worse, though I get your point

ConorMcPersonal committed:
Simplify plus remove key click

ConorMcPersonal commented:
I've simplified the control code and removed the key click - that made the controls feel a bit spongy to me and seemed to interfere with slot sounds.

I approve your changes

mcmenaminadrian commented:
I’ve just gone to bed! If you are happy with it please merge.

On Wed, 6 Dec 2023 at 22:02, ConorMcPersonal ***@***.***>
wrote:

> I've simplified the control code and removed the key click - that made the
> controls feel a bit spongy to me and seemed to interfere with slot sounds.
>
> I approve your changes
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ConorMcPersonal/Toaster/pull/75#issuecomment-1843761377>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACQK2K2FK5ZIPGOEVL3TRTYIDTRHAVCNFSM6AAAAABAIIU7DSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNBTG43DCMZXG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
